SetupDataPkg/Tools/GenCfgData_test.py

```diff
@@ -0,0 +1,529 @@
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.DOUBLE_KNOB')
+        val_str = '5.293845'
+        cdata.set_item_value(val_str, item=ret)
+        self.assertEqual(5.293845, ret['inst'].value)
+
+        settings = [] 
+        b64data = base64.b64encode(cdata.generate_binary_array())
+        settings.append(("Device.RuntimeData.RuntimeData", b64data.decode("utf-8")))
+        b64data = base64.b64encode(ydata.generate_binary_array())
+        settings.append(("Device.ConfigData.ConfigData", b64data.decode("utf-8")))
+        path = 'merged_xml_yml_bin.bin'
+        temp_file = path + '.tmp'
+        set_lib = SettingsXMLLib()
+        set_lib.create_settings_xml(
+            filename=temp_file, version=1, lsv=1, settingslist=settings
+        )
+
+        svd = ''
+
+        # To remove the line ends and spaces
+        with open(temp_file, "r") as tf:
+            for line in tf:
+                svd += line.strip().rstrip("\n")
+        os.remove(temp_file)
+
+        with open(path, "wb") as fd:
+            bins = b''
+            bins += ydata.generate_var_list_from_svd(svd)
+            bins += cdata.generate_binary_array()
+            fd.write(bins)
+
+        item = ydata.get_item_by_path('GFX_CFG_DATA.PowerOnPort0')
+        ydata.set_config_item_value(item, "1")
+
+        item = ydata.get_item_by_path('GFX_CFG_DATA.PowerOnPort0')
+        self.assertEqual("1", item['value'])
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.BOOLEAN_KNOB')
+        val_str = 'true'
+        cdata.set_item_value(val_str, item=ret)
+        self.assertEqual(ret['inst'].value, True)
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.DOUBLE_KNOB')
+        val_str = '5.293845'
+        cdata.set_item_value(val_str, item=ret)
+        self.assertEqual(5.293845, ret['inst'].value)
+
+        with open(path, "rb") as fd:
+            bin_data = bytearray(fd.read())
+
+        ydata.load_default_from_bin(bin_data, True)
+        cdata.load_default_from_bin(bin_data, True)
+        os.remove(path)
+
+        item = ydata.get_item_by_path('PLATFORM_CFG_DATA.PlatformName')
+        self.assertEqual("'PlatName'", item['value'])
+
+        item = ydata.get_item_by_path('GFX_CFG_DATA.PowerOnPort0')
+        self.assertEqual('0', item['value'])
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.INTEGER_KNOB')
+        self.assertEqual(ret['inst'].value, 100)
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.COMPLEX_KNOB1a.mode')
+        self.assertEqual(ret['inst'].value, 0)
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.BOOLEAN_KNOB')
+        self.assertEqual(False, ret['inst'].value)
+
+        ret = cdata.get_item_by_path('{FE3ED49F-B173-41ED-9076-356661D46A42}.DOUBLE_KNOB')
+        self.assertEqual(5.293845, ret['inst'].value)
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
nit: extra line at the end of a script file.

Reply:
You and flake8 both :). Fixed.